src/shm_ext.c

```diff
@@ -547,13 +557,13 @@ sr_shmext_change_sub_add(sr_conn_ctx_t *conn, sr_mod_t *shm_mod, sr_datastore_t
     sr_mod_change_sub_t *shm_sub;
     uint32_t i;
 
+    sr_shmext_change_sub_remove_dead(conn, shm_mod, ds);
+
```

Comment:
I am not sure such constructs really make sense, I always thought not. Meaning to hold only READ lock first because it is enough and later unlocking it and getting WRITE lock. Results in a more confusing code, more locks and may ultimately cause the operation to take longer because another session may get the WRITE lock and this one will have to wait for it. So when the operation is fast (traversing an array of module subscription for a DS), I would just stick with the WRITE lock to keep it organized, I do not believe this change made any relevant optimization. Similarly for further similar changes.

Reply:
I had 3 reasons for doing this:

1. All `sr_shmext___sub_del` and `sr_shmext__sub_stop` functions will be uniform in having READ remap + ext_lock.
2. Align with `sr_shmext_rpc_sub_remove_dead` which already takes the READ remap + ext_lock.
3. The operation here is not just traversing an array of module subscriptions, but doing a `sr_conn_is_alive()` check which does:
  a. `asprintf()` to get the malloc the connection file path string
  b. `sr_open()`
  c. `fcntl()`

I can leave this as it is, if you still prefer the previous code.

Reply:
Yeah, you are right there is more to do so okay, I suppose.